contents/blog/clickhouse-vs-bigquery.md

```diff
@@ -0,0 +1,168 @@
+
+![ClickHouse Cloud Architecture.png](../images/blog/clickhouse-vs-bigquery/clickhouse-cloud-architecture.png)
+
+ClickHouse Cloud is unquestionably modeled after BigQuery and GCP. It offers similar features, such as out-of-the-box integrations with ELT providers, an interactive SQL console, automated backups, caching, and automatic replication. 
+
+However, ClickHouse Cloud is still ***not*** a pure replacement for BigQuery. ClickHouse Cloud simply makes managing, scaling, and protecting a ClickHouse instance easy — it does ***not*** alter the way ClickHouse works, including what it is and isn’t good at. 
+
+> **Using ClickHouse + GCS:** It’s worth mentioning that another hybrid architecture exists — ClickHouse + Google Cloud Storage (GCS). Specifically, developers could replace ClickHouse’s in-unit storage with Google Cloud Storage instead. ClickHouse’s [integration with GCS](https://clickhouse.com/docs/en/integrations/gcs) effectively decouples storage and compute, and works great for businesses that already use GCS. 
+
+## Why ClickHouse is faster
+
+Data warehouses like BigQuery and ClickHouse are used to return analytical queries. Queries might look like “a list of average revenue for the last four years” or “the median household income”. Not only do these queries change on **every** data insert, but computing many values can draw a lot of memory and processing power. 
+
+The performance difference between BigQuery and ClickHouse can be immense. BigQuery can take dozens of seconds to execute a query. ClickHouse, if tuned correctly, can execute the same query on *terabytes* of data with sub-second performance. 
+
+The **"tuned”** bit is important — ClickHouse gives developers tooling to expedite expected queries. While BigQuery does some standard optimizations, like columnar storage (which we discuss in more detail in our [ClickHouse and Postgres comparison](/blog/clickhouse-vs-postgres)), ClickHouse goes further. In particular, ClickHouse accomplishes its “fighter jet” speeds by leveraging four techniques:
+
+1. Granulated storage
+2. Materialized views
+3. Specialized engines
+4. Vectorized query execution.
+
+### Granulated Storage
+
+Most databases store data indexed via primary keys. When searching for that data, such as a specific column, a typical database scans all data and return the result. This is sloooow. 
+
+ClickHouse achieves a magnificent speed boost by using granulated storage. Each ClickHouse table has a primary key defined as a tuple of multiple columns. Data is then sorted by those columns, first by the first entry, then by the second, and onwards. Afterwards, sorted entries are broken into granules, which are fixed length mini-tables, by default 8192 entries long. ClickHouse’s big advancement is that it exclusively saves just the **first** entry to each granule. 
```

Comment:
This would be a great spot to talk about how CH uses [Log Structured Merge Trees](https://en.wikipedia.org/wiki/Log-structured_merge-tree) (LSMT) to store data. This + the Granules makes CH very fast